client/src/Create.js

```diff
@@ -15,27 +15,65 @@ import styles from './css/create.module.css';
 function Create () {
 
     /* book review components */
-    // const [name, setName] = useState('');
-    // const [bookName, setBookName] = useState('');
-    // const [review, setReview] = useState('');
-    // const [categories, setCategories] = useState('');
+    const [user_name, setUserName] = useState('');
+    const [book_name, setBookName] = useState('');
+    const [text, setReview] = useState('');
+    const [categories, setCategories] = useState([]);
 
     /* other page related components */
-    // const [isPending, setIsPending] = useState(false);
+    const [isPending, setIsPending] = useState(false);
     const history = useHistory();
     const [validated, setValidated] = useState(false);
 
+
+    const handleChange = (e) => {
+        let field = e.target.name;
+        let value = e.target.value;
+
+        if (field === 'user') {
+            setUserName(value);
+        } else if (field === 'book') {
+            setBookName(value);
+        } else if (field === 'review') {
+            setReview(value);
+        } else {
+            const categoryArray = value.split(",");
+            setCategories(categoryArray);
+        }
+    }
+
+    async function createReview () {
+        const newReview = {user_name, book_name, text, categories};
+        const response = await fetch('https://bookworms-api.vercel.app/api/reviews', {
+            method: 'POST',
+            // convert the React state to JSON and send it as the POST body
+            headers: {'Content-Type': 'application/json'},
+            body: JSON.stringify(newReview)
+        })
+        // if the request wasn't succesful, throw an error for the user to know 
+        if (!response.ok) {
+            const message = `An error has occured: ${response.status}`;
+            throw new Error(message);
+        } else {
+            setIsPending(false);
+        }
+    }
+
     function handleSubmit(e) {
         const form = e.currentTarget;
+        // check for blank fields
         if (form.checkValidity() === false){
-            e.preventDefault();
             e.stopPropagation();
         } else {
-            e.preventDefault();
-            history.push('/');
+            setIsPending(true);
+            // slow down the switching back to home page for a little
+            setTimeout(() => {
+                createReview();
+                history.push('/');
+            }, 2000);
```

Comment:
I put it there because the button for "Submit" will be change to "Adding..." --> I was hoping to give the user a sign that we are processing their request, if that makes sense 😆

Reply:
Haha, OK, I give it a pass.